pkg/xds/bootstrap/generator.go

```diff
@@ -27,7 +27,7 @@ import (
 )
 
 type BootstrapGenerator interface {
-	Generate(ctx context.Context, request types.BootstrapRequest) (proto.Message, error)
+	Generate(ctx context.Context, request types.BootstrapRequest) (proto.Message, *configParameters, error)
```

Comment:
we need a better type here. configParameters is private, the interface should only return public types.